Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -57,14 +60,17 @@ async function run() {
             tl.setVariable(webAppUri, publishingProfile.destinationAppUrl);
         }
 
-        if(jsonVariableSubsFlag) { // (jsonVariableSubsFlag || variable substitution)
+        if(jsonVariableSubsFlag || xmlTransformsAndVariableSubstitutions) { // (jsonVariableSubsFlag || variable substitution)
             var folderPath = path.join(tl.getVariable('System.DefaultWorkingDirectory'), 'temp_web_package_folder');
             if(isFolderBasedDeployment) {
                 tl.cp(path.join(webDeployPkg, '/'), folderPath, '-rf', false);
```

Comment:
please change `path.join(webDeployPkg, '/')` to `path.join(webDeployPkg, '/*') `